File: src/main/java/org/hisp/dhis/rules/models/RuleActionUnsupported.java
package org.hisp.dhis.rules.models;

import com.google.auto.value.AutoValue;

import javax.annotation.Nonnull;

/**
 * @param data
 * @param content         a message to show to user
 *                        when an actionType is not supported
 * @param actionValueType name of the unsupported action.
 */
public record RuleActionUnsupported(
        @Nonnull String data,
        @Nonnull String content,
        @Nonnull String actionValueType
) implements RuleAction {

    @Nonnull
    public static RuleActionUnsupported create(@Nonnull String content, @Nonnull String actionValueType) {
        return new RuleActionUnsupported("", content, actionValueType);
    }
}

File: src/commonMain/kotlin/org/hisp/dhis/rules/models/RuleActionUnsupported.kt
package org.hisp.dhis.rules.models

/**
 * @param data
 * @param content         a message to show to user
 * when an actionType is not supported
 * @param actionValueType name of the unsupported action.
 */

data class RuleActionUnsupported(
    val content: String,
    val actionValueType: String,
    val data: String
) : RuleAction
